src/diff.rs

```diff
@@ -1,103 +1,252 @@
+            assert!(result.common.is_empty());
+            assert!(result.exclusive_left.is_empty());
+            assert!(result.exclusive_right.is_empty());
+        }
+
+        #[test]
+        fn map_diff_left_exclusive() {
+            let left: HashMap<&str, i32> = HashMap::from([("123", 2)]);
+            let right: HashMap<&str, i32> = HashMap::from([]);
+            let result = MapComparison::from_hash_maps(&left, &right);
+            assert!(result.common.is_empty());
+            assert_eq!(result.exclusive_left, vec![(&"123", &2)]);
+            assert!(result.exclusive_right.is_empty());
+        }
+
+        #[test]
+        fn map_diff_right_exclusive() {
+            let left: HashMap<&str, i32> = HashMap::from([]);
+            let right: HashMap<&str, i32> = HashMap::from([("123", 2)]);
+            let result = MapComparison::from_hash_maps(&left, &right);
+            assert!(result.common.is_empty());
+            assert!(result.exclusive_left.is_empty());
+            assert_eq!(result.exclusive_right, vec![(&"123", &2)]);
+        }
 
-        MapComparison {
-            exclusive_left,
-            exclusive_right,
-            different_values,
-            common,
+        #[test]
+        fn map_diff_common() {
+            let left: HashMap<&str, i32> = HashMap::from([("123", 2)]);
+            let right: HashMap<&str, i32> = HashMap::from([("123", 2)]);
+            let result = MapComparison::from_hash_maps(&left, &right);
+            assert_eq!(result.common, vec![(&"123", &2)]);
+            assert!(result.exclusive_left.is_empty());
+            assert!(result.exclusive_right.is_empty());
         }
     }
 }
 
-#[cfg(test)]
-mod tests {
-    use crate::diff::MapComparison;
-    use std::collections::HashMap;
+pub(crate) mod iter {
+    use std::fmt::Debug;
 
-    #[test]
-    fn diff_empty_maps() {
-        let left: HashMap<&str, i32> = HashMap::from([]);
-        let right: HashMap<&str, i32> = HashMap::from([]);
-        let result = MapComparison::from_hash_maps(&left, &right);
-        assert!(result.common.is_empty());
-        assert!(result.exclusive_left.is_empty());
-        assert!(result.exclusive_right.is_empty());
+    /// Differences between two Sequence-like structures.
+    pub(crate) struct SequenceComparison<T: PartialEq + Debug> {
+        pub(crate) order_preserved: bool,
+        pub(crate) exclusive_left: Vec<T>,
+        pub(crate) exclusive_right: Vec<T>,
     }
 
-    #[test]
-    fn map_diff_left_exclusive() {
-        let left: HashMap<&str, i32> = HashMap::from([("123", 2)]);
-        let right: HashMap<&str, i32> = HashMap::from([]);
-        let result = MapComparison::from_hash_maps(&left, &right);
-        assert!(result.common.is_empty());
-        assert_eq!(result.exclusive_left, vec![(&"123", &2)]);
-        assert!(result.exclusive_right.is_empty());
+    pub(crate) enum SequenceOrderComparison {
+        Relative,
+        Exact,
```

Comment:
SequenceComparisonMode?  
For element naming, `ExactlyOrdered` and `Unordered`?  If you have better idea, feel free to rename.

I could not get the meaning of `relative` in this context.  Though you use `strict` in method name, the name should align with the name of enum.